packages/shortest/src/browser/core/browser-tool.ts

```diff
@@ -389,6 +395,69 @@ export class BrowserTool extends BaseBrowserTool {
           }
         }
 
+        case 'sleep_milliseconds': {
+          const defaultDuration = 1000;
+          const maxDuration = 60000;
+          let duration = input.duration ?? defaultDuration;
+          
+          // Enforce maximum duration
+          if (duration > maxDuration) {
+            console.warn(`Requested sleep duration ${duration}ms exceeds maximum of ${maxDuration}ms. Using maximum.`);
+            duration = maxDuration;
+          }
+
+          // Convert to seconds for logging
+          const seconds = Math.round(duration / 1000);
+          console.log(`⏳ Waiting for ${seconds} second${seconds !== 1 ? 's' : ''}...`);
+          
+          await this.page.waitForTimeout(duration);
+          output = `Finished waiting for ${seconds} second${seconds !== 1 ? 's' : ''}`;
+          break;
+        }
+
+        case 'render_email': {
+          if (!this.mailosaurTool) {
+            if (!this.config.mailosaur) {
+              throw new ToolError('Mailosaur configuration required');
+            }
+            this.mailosaurTool = new MailosaurTool(this.config.mailosaur);
+          }
+
+          const newPage = await this.page.context().newPage();
+
+          try {
+            const email = await this.mailosaurTool.getLatestEmail();
+
+            // Render email in new tab
+            await newPage.setContent(email.html, {
+              waitUntil: 'domcontentloaded'
+            });
+
+            await newPage.waitForLoadState('load', {
+              timeout: 5000
+            }).catch(e => {
+              console.log('⚠️ Load timeout, continuing anyway');
+            });
+
+            // Switch focus
+            this.page = newPage;
+
+            output = `Email received successfully. Navigated to new tab to display email: ${email.subject}`;
+            metadata = {
+              window_info: {
+                title: email.subject,
+                content: email.html,
+                size: this.page.viewportSize() || { width: this.width, height: this.height }
+              }
+            };
+
+            break;
+          } catch (error) {
+            await newPage.close();
+            throw new ToolError(`Failed to render email: ${error}`);
+          }
+        }
+
```

Comment:
Implementation for render_email: This tool will grab the latest email from Mailosaur in HTML format and will render it in a new tab. It will then give the control back to AI to validate the email. 

Why render it in a new tab? 
Some users may want to test the links in the email. This gives AI a chance to interact with the email and click on it since it's in the browser. 

Just a note that this is indeterministic. Perhaps we could implement a mechanism to manually compare the content of the email and the received email in the inbox.